sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -200,6 +200,8 @@ object RemoveAliasOnlyProject extends Rule[LogicalPlan] {
         case plan: Project if plan eq proj => plan.child
         case plan => plan transformExpressions {
           case a: Attribute if attrMap.contains(a) => attrMap(a)
+          case b: Alias if attrMap.exists(_._1.exprId == b.exprId)
+            && b.child.isInstanceOf[NamedExpression] => b.child
```

Comment:
RemoveAliasOnlyProject will remove  `alias a#1 to a#2`, and `replace all a#2 with a#1`, so there is no `a#2` exists, If we do nothing for `alias a#1 to a#2`(not the same object with the removed one), it will cause the exception situation from `step 5` to `step 6` showed on the above comment.